gmso/abc/abstract_site.py

```diff
@@ -45,6 +51,14 @@ class Site(GMSOBase):
 
     label_: str = Field("", description="Label to be assigned to the site")
 
+    residue_index_: Optional[StrictInt] = Field(
```

Comment:
Rename it to `residue_number` maybe (Suggested by @justinGilmer). What do you think @bc118?